Include/cpython/object.h

```diff
@@ -446,6 +446,21 @@ PyAPI_FUNC(void) _PyObject_AssertFailed(
     int line,
     const char *function);
 
+/* Check if an object is consistent. For example, ensure that the reference
+   counter is greater than or equal to 1, and ensure that ob_type is not NULL.
+
+   Call _PyObject_AssertFailed() if the object is inconsistent.
+
+   If check_content is zero, only check header fields: reduce the overhead.
+
+   The function always return 1. The return value is just here to be able to
+   write:
```

Comment:
Why would you write `assert(_PyObject_CheckConsistency)` if it always returns 1? Wouldn't it be better off being named `AssertConsistent` if it's going to do the assert itself?

It looks like the other functions like this return 1 or 0 and the caller does the assert.

Reply:
> Why would you write assert(_PyObject_CheckConsistency) if it always returns 1?

It's a practical solution to add checks in debug mode which are omitted in release mode. It's shorter than:

```
#ifndef NDEBUG
_PyObject_CheckConsistency();
#endif
```

or

```
#ifdef Py_DEBUG
_PyObject_CheckConsistency();
#endif
```

By the way, I recall that someone reported an issue when Py_DEBUG is build *without* assertions (with NDEBUG defined).

Reply:
Ah, got it, that makes sense. But if this is internal, why not just offer a macro that is defined to blank?

On Windows, `Py_DEBUG` is inferred from `_DEBUG`, which is the opposite of `NDEBUG` by default (but I'm not totally sure which one modifies `assert`). We should definitely use `Py_DEBUG` within the codebase for our own checks like this, and probably people should not set it directly? Or maybe we need to add some extra validation in `PC/pyconfig.h` to fail the build if it's inconsistent. (Let's worry about this when someone files a bug, rather than here :) )